api/pkg/transformer/feast/feature_retriever.go

```diff
@@ -106,12 +117,39 @@ type Options struct {
 	FeastClientErrorPercentThreshold  int           `envconfig:"FEAST_HYSTRIX_ERROR_PERCENT_THRESHOLD" default:"25"`
 }
 
-func (o Options) IsServingURLSupported(url string) bool {
-	for _, supportedURL := range o.ServingURLs {
-		if supportedURL == url {
+func (o Options) IsFeastConfigUsingCorrectSource(featureTableCfg *spec.FeatureTable) bool {
```

Comment:
I think this should be part of `validator.go` and the method should accept the `Options` as parameter not as receiver.  i.e.
```
func IsFeastConfigUsingCorrectSource(option Options, featureTableCfg *spec.FeatureTable) bool {
 ...
}
```

Reply:
The name also need to be improved to reflect better about what the function does.